src/vs/platform/actions/common/actions.ts

```diff
@@ -103,6 +103,9 @@ export class MenuId {
 	static readonly SCMSourceControl = new MenuId('SCMSourceControl');
 	static readonly SCMTitle = new MenuId('SCMTitle');
 	static readonly SearchContext = new MenuId('SearchContext');
```

Comment:
Seems that we currently have one menu for all search row types, and actions are contributed to this with a context key to enable them for certain types. But for the inline actions, there is one menu per type. I think these should match- either one menu for inline actions, or multiple menus for context keys.

Reply:
if we have one menu, it's a little tricky to determine the match type to replace just using existing context keys since the one we're targeting isn't always the focused one, so we can't use keys like `FileFocusKey`. We'd need to find some way to see what type of match is being "highlighted" aka the action bar is being shown for. Is there an easy way for that? 

By having different menus for each match type, we don't need to worry about differentiating the match type by context.

Reply:
Yes, I don't really like the way it's done in search where the keys are tied to "focus", that's not how I've done it other places. Essentially you can create a child ContextKeyService (or overlay) for each row and set the type keys in it. Your options for which type keys to use look like this:

- Reuse the keys like "FileFocusKey". Now they are technically not about "focus" but that's ok. They are about some concept of the "current" item and should be used the same
- Create keys for each type that don't have "focus" in the name
- Create one key that has a string type, for the current item type. Example: https://github.com/microsoft/vscode/blob/eddb3e5c4a3a99547780c96f0b12a78aa416b087/src/vs/workbench/contrib/notebook/browser/controller/editActions.ts#L55

I think I would go with the first option, since I think we generally use the same context key for focus-driven things as per-item menus, but I'd probably review how those keys are used and maybe check what some other trees do as well.

Reply:
If we use the first option to re-use the keys for focus, it might be hard to determine which entry is focused and which one is being hovered, for example. Technically, to me, they're both being "highlighted".

<img width="536" alt="image" src="https://user-images.githubusercontent.com/31675041/200651097-c178af19-142e-4740-ab14-bbcde04184c9.png">

In that case, I guess the second option would be better, adding something like a "hovered" key?

Reply:
Also, I'm a little afraid that this'll be adding more changes to an already-large PR. But if it's preferred to be done all at once, I get that too